File: typescript/src/modules/weather.ts
#!/usr/bin/env node

import * as env from '../env';
import {getIPAddress, getLatLonCoords, truncateDecimals} from './utility';
import axios from 'axios';

export async function weatherReport() {
    let addr = await getIPAddress();
    let coords = await getLatLonCoords(addr);
    
    const url = `https://api.openweathermap.org/data/2.5/weather?lat=${coords[0]}&lon=${coords[1]}&appid=${env.OPEN_WEATHER_API_KEY}`

    const response = await axios.get(url);

    const temp = (response.data.main.temp - 273) * 1.8 + 32;

    const report = {
        description: response.data.weather[0].description,
        temp: truncateDecimals(temp, 2),
        humidity: response.data.main.humidity,
    };

    console.log(report);
}

File: python/src/modules/weather.py
import modules.utility as utility
import os
import pprint


def setUpApiCall():
    addr = utility.getIpAddress()
    coords = utility.getLatLonCoords(addr)

    return coords

def main():
    excludes = "minutely,current"
    units = "imperial"
    apiKey = os.getenv('OPEN_WEATHER_API_KEY')
    coords = setUpApiCall()
    url = f'https://api.openweathermap.org/data/2.5/onecall?lat={coords[0]}&lon={coords[1]}&exclude={excludes}&units={units}&appid={apiKey}'
    
    response = utility.apiGet(url)
    pprint.pp(response)